mithril-client/src/commands/download.rs

```diff
@@ -36,8 +41,18 @@ impl DownloadCommand {
             .map_err(|e| format!("configuration deserialize error: {e}"))?;
         debug!("{:?}", config);
         let runtime = Runtime::new(config.network.clone());
-        let aggregator_handler =
-            AggregatorHTTPClient::new(config.network.clone(), config.aggregator_endpoint);
+        // TODO: This does not allow the client to handle an upgraded API version of a new supported era after the switch
+        // In order to do so, we should retrieve the list of supported versions from the API Version provider and test them sequentially until one hopefully succeeds
+        let era_checker = Arc::new(EraChecker::new(
```

Comment:
Maybe it would be useful to get rid of the `EraChecker` dependency which is not needed in this case to determine the API version to use.